Add responsive styling for individual /integrations/doc/ pages.

sameerchoubey opened the pull request:
@timabbott @rishig 

This is ready for review.
![Screenshot from 2019-03-21 14-47-18](https://user-images.githubusercontent.com/25278879/54742791-590c8780-4be8-11e9-827e-f3088e20e909.png)

I went through all of this - https://chat.zulip.org/#narrow/stream/137-feedback/topic/Zulip.20Integration.20Docs.20on.20Mobile/near/708632 

The reason why I didn't use a logo because there was a particular a `div` that contained the logo along with the box and all and it was appearing on the web version of the page and by default, it was hidden for mobile pages(`display:none`). Making any changes to that div was altering the whole design. So, I instead copied the `img` tag and tried using the place, I wanted the logo to appear. But, it was not working. So, I opted to use this.

I think the current design is simple and does the work. Let me know.
This PR closes the https://github.com/zulip/zulip/issues/11701

timabbott commented:
@sameerchoubey I can't really review this without clear explanation for why the changes you've made to various styles that are potentially used on other pages won't break other widgets

Also, your commit history is confusing.  Check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages

sameerchoubey commented:
I restructured the code and it looks a lot cleaner now. 

The problem was caused by the `.integration-instructions` which had `margin-left: 220px;`.  So, I created a media query (`media (min-width: 768px)`) and moved this class into it. So, the margin will only be applicable for screens greater than 768px. 

Next, the `#integration-list-link` ID had top and left values assigned which was for the desktop version (i.e. to accommodate the logo, categories, etc.), so, I again moved a copy of the ID a to the media query and lastly, removed the top and left attributes from the existing ID (which were creating problems for the mobile view). 

I think that this should not break any other pages. Also, the ID seems to be for the specific links present on this page only.
It looks as below.
![Screenshot from 2019-03-22 17-30-48](https://user-images.githubusercontent.com/25278879/54821485-49647000-4cc8-11e9-9985-50a87738fb3e.png)

@timabbott Please review.

AsociTon commented:
tested it on my local machine the only error I get is the mis-alignment of the `integration-list-link` posted below, also please add a bit more detail to the commit message, you can check our commit guidelines at https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages
![Screenshot from 2019-04-15 20-05-57](https://user-images.githubusercontent.com/41135524/56141411-63892980-5fba-11e9-82eb-5f7b37c2fb52.png)

* One more thing would be, in the discussion the suggestion was to include the orgs logo and to remove the border surrounding it, please confirm it in the conversation and add it too if required.

sameerchoubey commented:
@AsociTon

So, I just re-checked on my local environment. The design seems fine. 
![Screenshot from 2019-04-15 20-35-16](https://user-images.githubusercontent.com/25278879/56144018-3ab76300-5fbf-11e9-953f-cd8ac3062b2c.png)

I wanted to keep it simple, so, I didn't use the logo because we're basically on the same page. Also, there was already a topic open in the `feedback` thread, so, I used it since this PR was a quick fix to the issue.

Commit message seems fine and is on the same line as few of my last merged PR in design section.

sameerchoubey commented on static/styles/portico.scss:
```diff
@@ -1233,8 +1243,6 @@ input#terminal:checked ~ #tab-terminal {
 
 #integration-list-link {
     position: absolute;
-    top: 200px;
```
Okay. So, the thing is that, this part of the CSS is still there. I just have shifted it to the`@media (min-width: 768px)` which basically means that this all values will apply to screen above 768px. 

I've done this for obvious reasons as mentioned in the other comment on this PR.

sameerchoubey replied:
Okay. So, the thing is that, this part of the CSS is still there. I just have shifted it to the`@media (min-width: 768px)` which basically means that this all values will apply to screen above 768px. 

I've done this for obvious reasons as mentioned in the other comment on this PR.

AsociTon replied:
ohk @sameerchoubey,  I used my master branch for the testing and seems like the error was rising because of the browser cache, idk for sure but rechecked after re-running the browser and reloading with `ctrl + F5` made it work.
P.S. : everything works fine :+1:

AsociTon commented:
@zulipbot abandon
> The reason why I didn't use a logo because there was a particular a `div` that contained the logo along with the box and all and it was appearing on the web version of the page and by default, it was hidden for mobile pages(`display:none`). Making any changes to that div was altering the whole design. So, I instead copied the `img` tag and tried using the place, I wanted the logo to appear. But, it was not working. So, I opted to use this.
> 
> I think the current design is simple and does the work. Let me know.
> This PR closes the #11701

@sameerchoubey did you try setting the border to 0 in case of mobile view (i.e. less than 768), It will get rid of the border and you can proceed with setting a flex for the logo and the org's name in the same line for obtaining suggested design https://chat.zulip.org/#narrow/stream/101-design/topic/Zulip.20Integration.20Docs.20on.20Mobile/near/733194 :+1:

timabbott commented:
I cleaned this up a bit and merged it as d711b1c8ba270d36369007df604fee1762882104; thanks @sameerchoubey for doing this!  I think we could make the headings for those pages a lot nicer, and probably should in follow-up work, but this at least makes the page non-broken.  I also added 734c18f35644adb9a1b309edbd8dcda13662e418 to fix a major visual glitch with the number bullets.  

Thanks @sameerchoubey and @AsociTon!

timabbott commented:
I also opened https://github.com/zulip/zulip/issues/12365 for the follow-up work.  @vrongmeal as a sidenote, there's a bunch of sections in our portico page CSS that is ripe for SCSS stuff; we might want to wait to work on that until we've done some smaller files, though, since they are kinda huge.

vrongmeal commented:
> we might want to wait to work on that until we've done some smaller files, though, since they are kinda huge.

Yeah, sure :)
